### ISSUE TITLE
Language Service and StartupProjectRegistrar are broken in 15.7 builds

### ISSUE THREAD

**Issue by davkean** [dotnet/project-system#3341]
```
Microsoft.VisualStudio.ProjectSystem.VS.LanguageServices.UnconfiguredProjectContextProvider.ctor(projectGuidService): expected exactly 1 export of Microsoft.VisualStudio.ProjectSystem.VS.IProjectGuidService2 but found 2.
    Microsoft.VisualStudio.Project.VisualC.VCProjectEngine.VCProjectShim
    Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.ProjectNode
   part definition Microsoft.VisualStudio.ProjectSystem.VS.LanguageServices.UnconfiguredProjectContextProvider

Microsoft.VisualStudio.ProjectSystem.VS.LanguageServices.UnconfiguredProjectContextProvider.ctor(projectGuidService): is not assignable from exported MEF value Microsoft.VisualStudio.Project.VisualC.VCProjectEngine.VCProjectShim.
   part definition Microsoft.VisualStudio.ProjectSystem.VS.LanguageServices.UnconfiguredProjectContextProvider

Microsoft.VisualStudio.ProjectSystem.VS.Debug.StartupProjectRegistrar.ctor(projectGuidService): expected exactly 1 export of Microsoft.VisualStudio.ProjectSystem.VS.IProjectGuidService2 but found 2.
    Microsoft.VisualStudio.Project.VisualC.VCProjectEngine.VCProjectShim
    Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.ProjectNode
   part definition Microsoft.VisualStudio.ProjectSystem.VS.Debug.StartupProjectRegistrar

Microsoft.VisualStudio.ProjectSystem.VS.Debug.StartupProjectRegistrar.ctor(projectGuidService): is not assignable from exported MEF value Microsoft.VisualStudio.Project.VisualC.VCProjectEngine.VCProjectShim.
   part definition Microsoft.VisualStudio.ProjectSystem.VS.Debug.StartupProjectRegistrar

----- CompositionError level 2 ------
Microsoft.VisualStudio.ProjectSystem.LanguageServices.LanguageServiceHost.ctor(contextProvider): expected exactly 1 export of Microsoft.VisualStudio.ProjectSystem.LanguageServices.IProjectContextProvider but found 0.
   part definition Microsoft.VisualStudio.ProjectSystem.LanguageServices.LanguageServiceHost

----- CompositionError level 3 ------
Microsoft.VisualStudio.ProjectSystem.VS.LanguageServices.VsContainedLanguageComponentsFactory.ctor(languageServiceHost): expected exactly 1 export of Microsoft.VisualStudio.ProjectSystem.LanguageServices.ILanguageServiceHost but found 0.
   part definition Microsoft.VisualStudio.ProjectSystem.VS.LanguageServices.VsContainedLanguageComponentsFactory

Microsoft.VisualStudio.ProjectSystem.VS.Properties.ProjectFileOrAssemblyInfoPropertiesProvider.ctor(languageServiceHost): expected exactly 1 export of Microsoft.VisualStudio.ProjectSystem.LanguageServices.ILanguageServiceHost but found 0.
   part definition Microsoft.VisualStudio.ProjectSystem.VS.Properties.ProjectFileOrAssemblyInfoPropertiesProvider

Microsoft.VisualStudio.ProjectSystem.VS.Build.LanguageServiceErrorListProvider.ctor(host): expected exactly 1 export of Microsoft.VisualStudio.ProjectSystem.LanguageServices.ILanguageServiceHost but found 0.
   part definition Microsoft.VisualStudio.ProjectSystem.VS.Build.LanguageServiceErrorListProvider

Microsoft.VisualStudio.ProjectSystem.Managed.VS.EditAndContinue.EditAndContinueProvider.ctor(host): expected exactly 1 export of Microsoft.VisualStudio.ProjectSystem.LanguageServices.ILanguageServiceHost but found 0.
   part definition Microsoft.VisualStudio.ProjectSystem.Managed.VS.EditAndContinue.EditAndContinueProvider

Microsoft.VisualStudio.ProjectSystem.VS.LanguageServices.ProjectContextCodeModelProvider.ctor(languageServiceHost): expected exactly 1 export of Microsoft.VisualStudio.ProjectSystem.LanguageServices.ILanguageServiceHost but found 0.
   part definition Microsoft.VisualStudio.ProjectSystem.VS.LanguageServices.ProjectContextCodeModelProvider


```

**Comment by davkean** [comment]
Fixed in 15.7.

**Comment by jcouv** [comment]
Confirmed. I was getting a popup with logs related to this in `Version 15.7.0 Preview 1.0 [27513.1.d15.7]`, but it seems to be resolved in `Version 15.7.0 Preview 2.0 [27517.0.d15.7]`.
Thanks

![image](https://user-images.githubusercontent.com/12466233/37670065-9970bcee-2c25-11e8-8baa-5810433c98ca.png)

That log file contains:
```
=====================
3/20/2018 9:59:28 AM
Recoverable
Microsoft.Assumes+InternalErrorException: An internal error occurred. Please contact Microsoft Product Support Services.
   at Microsoft.Assumes.Fail(String message, Boolean showAssert)
   at Microsoft.VisualStudio.ProjectSystem.VS.Debug.StartupProjectRegistrar.<InitializeCoreAsync>d__13.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.ProjectSystem.OnceInitializedOnceDisposedAsync.<<-ctor>b__4_0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Threading.AsyncLazy`1.<>c__DisplayClass13_1.<<GetValueAsync>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.ProjectSystem.UnconfiguredProjectImpl.AutoLoadMethodStateMachine.<<StartExecution>b__8_0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.ProjectSystem.OnceInitializedOnceDisposedAsync.<<InitializeAsync>b__27_0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.VisualStudio.Telemetry.WindowsErrorReporting.WatsonReport.GetClrWatsonExceptionInfo(Exception exceptionObject)
===================
```

**Comment by davkean** [comment]
Yep that was a separate bug: https://github.com/dotnet/project-system/issues/3338, which was also fixed.
